### PR TITLE
MGMT-10754: remove unused rootfs_url from OS images

### PR DESCRIPTION
**Pull request by danielerez**
Following the change in assisted-image-service to use boot-artifacts API for the fetching rootfs:
https://github.com/openshift/assisted-image-service/pull/83, removing rootfs_url from OS images.
The os-image swagger object wasn't exposed to the API, but just as a model internally. So removing the rootfs_url property should be safe.
Note: pending https://github.com/openshift/assisted-service/pull/3957

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @carbonin 
/cc @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: api/v1beta1/agentserviceconfig_types.go [hunk 1]
```diff
@@ -32,8 +32,6 @@ type OSImage struct {
 	Version string `json:"version"`
 	// Url specifies the path to the Operating System image.
 	Url string `json:"url"`
-	// rootFSUrl specifies the path to the root filesystem.
-	RootFSUrl string `json:"rootFSUrl"`
```
Comment (carbonin) [inline api/v1beta1/agentserviceconfig_types.go:36]: I'm not sure we can remove this.

This is a breaking change and will prevent our operator from upgrading properly.

Comment (danielerez) [reply api/v1beta1/agentserviceconfig_types.go:36]: ok, we can keep it for now then. do you know if we could remove it later without breaking upgradability?

Comment (carbonin) [reply api/v1beta1/agentserviceconfig_types.go:36]: Not without an api version change I think.
cc @avishayt @mhrivnak

Comment (avishayt) [reply api/v1beta1/agentserviceconfig_types.go]: Where is os-images used?  It's not in an API from what I can see.  Is it stored in the DB and that's the issue?

Comment (carbonin) [reply api/v1beta1/agentserviceconfig_types.go]: It's a part of the agentserviceconfig CR.

Comment (carbonin) [reply api/v1beta1/agentserviceconfig_types.go]: Here https://github.com/openshift/assisted-service/blob/86037f622038d984fffa517dc2b4f4df97667f7a/api/v1beta1/agentserviceconfig_types.go#L85

Comment (avishayt) [reply api/v1beta1/agentserviceconfig_types.go]: I think we could still remove it from the swagger and DB, and mark it in the agentserviceconfig as ignored, to be removed in the next API version.

Comment (danielerez) [reply api/v1beta1/agentserviceconfig_types.go]: Sure, marked RootFSUrl as deprecated in agentserviceconfig_types then.

**Review by carbonin — APPROVED** [review]

**Pull request merged**